### PR TITLE
gPRC swift protobuf easy installation + easy file generataions

### PR DESCRIPTION
**Pull request by El-Moatasem**
1. Adding Easy Installation Using Homebrew Package 
3. Adding Automatic Files Generation Script in Build Phases Xcode

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,6 +144,23 @@ the following line to your `Podfile`:
...
+cd "${SRCROOT}/${TARGET_NAME}/Data/gRPC_Schema"
+currentPath = "${SRCROOT}/${TARGET_NAME}/Data/gRPC_Schema"
+for i in $(ls | egrep '\.proto$'); 
+    do echo "$i";
+    protoc "$i" \
+             --swift_out=Visibility=Public:. \
+             --grpc-swift_out=Visibility=Public,Client=true,Server=false:.;
+    
+done; 
+```
```
Comment (glbrntt) [inline README.md:163]: I don't think we should include this section: build phase scripts aren't an unknown feature and the script itself is fairly simple. It's also highly likely to be different for different folks (different paths to .proto files, different plugin options and so on).

**Review by glbrntt — CHANGES_REQUESTED** [review]
Thanks for your patch @El-Moatasem, I left a few comments inline.

File: README.md (hunk 1 above)
Comment (El-Moatasem) [reply README.md:163]: Ok Great
I think it depends on your usage and it's not complicated to build 
I will remove it 
Thanks

File: README.md [hunk 2]
```diff
@@ -17,6 +17,17 @@ all four gRPC API styles (Unary, Server Streaming, Client Streaming, and
 Bidirectional Streaming) and connections can be made either over secure (TLS) or
 insecure channels.
 
+
+## Homebrew (Easy Installation)
```
[the remark this answers is not in the document]
Comment (El-Moatasem) [reply README.md:21]: Ok Sure
I will move it there "Getting the protoc plugins" section 
Thanks

File: README.md [hunk 3]
```diff
@@ -17,6 +17,17 @@ all four gRPC API styles (Unary, Server Streaming, Client Streaming, and
...
+1. Installing Swift-Protobuf : 
+```bash
+    $ brew install swift-protobuf
+```
+2. Installing GRPC-Swift :
```
[the remark this answers is not in the document]
Comment (El-Moatasem) [reply README.md]: Done

File: README.md [hunk 4]
```diff
@@ -17,6 +17,17 @@ all four gRPC API styles (Unary, Server Streaming, Client Streaming, and
 Bidirectional Streaming) and connections can be made either over secure (TLS) or
 insecure channels.
 
+
+## Homebrew (Easy Installation)
+1. Installing Swift-Protobuf :
```
[the remark this answers is not in the document]
Comment (El-Moatasem) [reply README.md]: Done

File: README.md [hunk 5]
```diff
@@ -17,6 +17,8 @@ all four gRPC API styles (Unary, Server Streaming, Client Streaming, and
 Bidirectional Streaming) and connections can be made either over secure (TLS) or
 insecure channels.
 
+
+
```
Comment (glbrntt) [inline README.md]: Lets remove these new blank lines

File: README.md [hunk 6]
```diff
@@ -131,6 +133,17 @@ the following line to your `Podfile`:
     pod 'gRPC-Swift-Plugins'
 ```
 
+Homebrew (Easy Installation)
```
Comment (glbrntt) [inline README.md]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -131,6 +133,17 @@ the following line to your `Podfile`:
...
+The plugins are available from homebrew and can be installed with :
```
Comment (glbrntt) [inline README.md:136]:
```suggestion
The plugins are available from [homebrew](https://brew.sh) and can be installed with:
```

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (El-Moatasem) [reply README.md]: Done

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (El-Moatasem) [reply README.md]: Done

File: README.md [hunk 8]
```diff
@@ -131,6 +133,18 @@ the following line to your `Podfile`:
...
 ```
 
+#### Homebrew (Easy Installation) :
+
+The plugins are available from homebrew and can be installed with :
+1. Installing Swift Protobuf : (swift-protobuf is installed as a dependency of grpc-swift)
+```bash
+    $ brew install swift-protobuf
+```
+2. Installing gRPC Swift :
```
Comment (glbrntt) [inline README.md:141]:
```suggestion

```

File: README.md [hunk 9]
```diff
@@ -131,6 +133,18 @@ the following line to your `Podfile`:
...
+#### Homebrew (Easy Installation) :
+
+The plugins are available from homebrew and can be installed with :
+1. Installing Swift Protobuf : (swift-protobuf is installed as a dependency of grpc-swift)
+```bash
+    $ brew install swift-protobuf
+```
+2. Installing gRPC Swift : 
+```bash
+    $ brew install grpc-swift
```
Comment (glbrntt) [inline README.md:143]:
```suggestion
    $ brew install swift-protobuf grpc-swift
```

File: README.md [hunk 10]
```diff
@@ -131,6 +131,18 @@ the following line to your `Podfile`:
     pod 'gRPC-Swift-Plugins'
 ```
 
+#### Homebrew (Easy Installation) :
```
Comment (glbrntt) [inline README.md:134]:
```suggestion
#### Homebrew
```

Comment (El-Moatasem) [reply README.md]: Done

File: README.md (hunk 9 above)
Comment (El-Moatasem) [reply README.md]: Done

File: README.md (hunk 8 above)
Comment (El-Moatasem) [reply README.md]: Done

File: README.md (hunk 7 above)
Comment (El-Moatasem) [reply README.md]: Done

File: README.md [hunk 11]
```diff
@@ -131,6 +131,14 @@ the following line to your `Podfile`:
     pod 'gRPC-Swift-Plugins'
 ```
 
+#### Homebrew :
```
Comment (glbrntt) [inline README.md:134]:
```suggestion
#### Homebrew
```

File: README.md [hunk 12]
```diff
@@ -131,6 +131,14 @@ the following line to your `Podfile`:
...
+
+The plugins are available from [homebrew](https://brew.sh) and can be installed with:
+1. Installing Swift Protobuf / gRPC Swift :
```
Comment (glbrntt) [inline README.md:137]:
```suggestion

```

Comment (El-Moatasem) [reply README.md]: Done

File: README.md (hunk 11 above)
Comment (El-Moatasem) [reply README.md]: Done

File: README.md [hunk 13]
```diff
@@ -131,6 +131,13 @@ the following line to your `Podfile`:
...
 ```
 
+#### Homebrew
+
+The plugins are available from [homebrew](https://brew.sh) and can be installed with:
+```bash
+    $ brew install swift-protobuf grpc-swift
+```
+
 The plugins are available in the `Pods/gRPC-Swift-Plugins/` folder afterwards.
```
Comment (glbrntt) [inline README.md:141]: Ah, this line has escaped from the section above, it should come before the `#### Homebrew` line.

**Review by glbrntt — APPROVED** [review]
Great, thank you!

File: README.md (hunk 13 above)
Comment (El-Moatasem) [reply README.md]: Yup
You are right!

Done

**Pull request merged**